src/applications/mhv/medical-records/util/helpers.js

```diff
@@ -194,3 +195,56 @@ export const getNameDateAndTime = user => {
     .format('M-D-YYYY_hhmmssa')
     .replace(/\./g, '')}`;
 };
+
+/**
+ * Custom hook for automatic data fetching at regular intervals.
+ *
+ * @param {function} dispatch - The Redux dispatch function.
+ * @param {function} fetchDataFunction - The function responsible for fetching data.
+ * @throws {Error} - Throws an error if dispatch or fetchDataFunction is not provided.
+ */
+export const useAutoFetchData = (dispatch, fetchDataFunctionFactory) => {
+  const INTERVAL = 2 * 60 * 1000;
+  if (
+    typeof dispatch !== 'function' ||
+    typeof fetchDataFunctionFactory !== 'function'
+  ) {
+    throw new Error(
+      'useAutoFetchData requires dispatch and fetchDataFunctionFactory as arguments.',
+    );
+  }
+
+  const lastFetchTime = useRef(0);
+
+  const fetchData = useCallback(
+    () => {
+      const fetchDataFunction = fetchDataFunctionFactory(); // Call the factory function to get the fetch function
+      if (typeof fetchDataFunction === 'function') {
+        dispatch(fetchDataFunction);
+        lastFetchTime.current = Date.now();
+      }
+    },
+    [dispatch, fetchDataFunctionFactory],
+  );
+
+  useEffect(
+    () => {
+      const timerId = setInterval(() => {
+        const currentTime = Date.now();
+
+        if (currentTime - lastFetchTime.current >= INTERVAL) {
+          fetchData();
+        }
+      }, INTERVAL);
```

Comment:
I think there was some confusion. The idea here was not to set an interval timer and auto-fetch the data every X minutes. Instead, we want to fetch the data on-request, but ONLY if it's been more than two minutes since the last time the data was fetched.

So basically:
1. The user browses to Allergies List View
2. Has it been more then two minutes since we fetched allergies list?
3. If yes, fetch the allergies list.
4. If no, do nothing.

To accomplish this, you are going to have to keep track of the last time each refresh happened for each domain.

Also, it just occurred to me that we do NOT need to do this for detail views, only for list views.